homeassistant/components/fritz/switch.py

```diff
@@ -0,0 +1,645 @@
+
+    async def _async_switch_on_off_executor(self, turn_on: bool) -> None:
+        """Handle deflection switch."""
+        await async_service_call_action(
+            self._fritzbox_tools,
+            "X_AVM-DE_OnTel",
+            "1",
+            "SetDeflectionEnable",
+            NewDeflectionId=self.id,
+            NewEnable="1" if turn_on else "0",
+        )
+
+
+class FritzBoxProfileSwitch(FritzBoxBaseSwitch, SwitchEntity):
+    """Defines a FRITZ!Box Tools DeviceProfile switch."""
+
+    def __init__(
+        self, fritzbox_tools: FritzBoxTools, device_friendly_name: str, profile: str
+    ) -> None:
+        """Init Fritz profile."""
+        self._fritzbox_tools: FritzBoxTools = fritzbox_tools
+        self.profile = profile
+
+        switch_info = SwitchInfo(
+            description=f"Profile {profile}",
+            friendly_name=device_friendly_name,
+            icon="mdi:router-wireless-settings",
+            type=SWITCH_TYPE_DEVICEPROFILE,
+            callback_update=self._async_fetch_update,
+            callback_switch=self._async_switch_on_off_executor,
+        )
+        super().__init__(self._fritzbox_tools, device_friendly_name, switch_info)
+
+    async def _async_fetch_update(self) -> None:
+        """Update data."""
+        try:
+            status = await self.hass.async_add_executor_job(
+                self._fritzbox_tools.fritz_profiles[self.profile].get_state
+            )
+            _LOGGER.debug(
+                "Specific %s response: get_State()=%s",
+                SWITCH_TYPE_DEVICEPROFILE,
+                status,
+            )
+            if status == "never":
+                self._attr_is_on = False
+                self._is_available = True
+            elif status == "unlimited":
+                self._attr_is_on = True
+                self._is_available = True
+            else:
+                self._is_available = False
+        except Exception:  # pylint: disable=broad-except
+            _LOGGER.error("Could not get %s state", self.name, exc_info=True)
+            self._is_available = False
+
+    async def _async_switch_on_off_executor(self, turn_on: bool) -> None:
+        """Handle profile switch."""
+        state = "unlimited" if turn_on else "never"
+        await self.hass.async_add_executor_job(
+            self._fritzbox_tools.fritz_profiles[self.profile].set_state, state
+        )
+
+    @property
+    def entity_registry_enabled_default(self) -> bool:
+        """Return if the entity should be enabled when first added to the entity registry."""
+        return False
+
+
+class FritzBoxWifiSwitch(FritzBoxBaseSwitch, SwitchEntity):
+    """Defines a FRITZ!Box Tools Wifi switch."""
+
+    def __init__(
+        self,
+        fritzbox_tools: FritzBoxTools,
+        device_friendly_name: str,
+        network_num: int,
+        network_name: str,
+    ) -> None:
+        """Init Fritz Wifi switch."""
+        self._fritzbox_tools: FritzBoxTools = fritzbox_tools
```

Comment:
Shouldn't need to be typed since mypy should figure it out from the constructor

Reply:
Removed